src/app/clusters/ota-requestor/OTARequestorDriver.h

```diff
@@ -24,19 +24,80 @@
 
 #pragma once
 
+#include <protocols/bdx/BdxMessages.h>
+
+namespace chip {
+
+// The set of parameters needed for starting a BDX download.
+struct BdxDownloadParameters
+{
+    uint32_t delayedActionTime;
+    chip::CharSpan imageURI;
+    uint32_t softwareVersion;
+    chip::CharSpan softwareVersionString;
+    chip::ByteSpan updateToken;
+    bool userConsentNeeded;
+    chip::ByteSpan metadataForRequestor;
+};
```

Comment:
How does the `ResumeImageDownload()` figure out from which offset to resume the download? Or should this have a parameter for offset where the application can tell from where to resume?

Reply:
Here's how we envision this:
For a download that needs to be resumed the application/platform maintains the offset internally. The ResumeImageDownload() does not contain the offset, the Requestor starts the download as any other download. Now, after the first block is passed to the platform in ProcessBlock() the platforms responds with a SkipData() passing the internally saved offset. This will generate a BlockQueryWithSkip message in the BDX session with the Provider and the download will continue from that offset.

Reply:
Thanks, understood.